Authentication and QOL changes

ShadowEO opened the pull request:
- Added authentication as per my request #6
- Moved Settings dialog into it's own menu option to help facilitate saving settings.
- Added rudimentary auto-login feature.

Currently the password is saved in plain-text in the client storage (whether this be on the server, or in the browser). This at least satisfies the use-case of authenticated Xpra use, if someone can provide a more secure way to store/retrieve the password, I'd be open to it.

(Recreated on request)

ShadowEO committed:
Added authentication, use of settings provider, and added rudimentary autologin; moved Settings dialog into separate menu option.

ShadowEO committed:
Updated password field in settings dialog to use password field type

ShadowEO commented:
Hopefully this PR works better for you. Great project btw, combining it with Xpra makes it exceptionally powerful. You may be seeing more contributions from me in the future as I attempt to make OS.js suit my needs!

andersevenrud commented:
Thanks!

And it seems like the workflow triggered this time :tada:

ShadowEO commented:
I see some linting issues, I'll work to fix those when I get back to my workstation

andersevenrud commented:
Yup. I'll finish up the review once you get those fixed :smile:

ShadowEO committed:
Hopefully fixed linting issues!

ShadowEO commented:
Alright, I fixed everything eslint complained about though there was one line that threw a warning (which was part of the original code, I didn't touch the websocket parts).

andersevenrud commented on index.js:
```diff
@@ -38,10 +38,15 @@ const register = (core, args, options, metadata) => {
-    uri: 'ws://localhost:10000',
-    sound: false
+  const proc = core.make('osjs/application', {args, options, metadata});
+  let defaultOptions = {
```
Revert this back to a `const`.

andersevenrud commented on index.js:
```diff
@@ -108,14 +149,27 @@ const register = (core, args, options, metadata) => {
+            createConnectionDialog(options => {
+              proc.settings = options;
+              proc.saveSettings();
+              defaultOptions = options;
```
Then change this to `Object.assign(defaultOptions, options)`

ShadowEO committed:
Updating index.js as per requested upstream

Updating the application code with changes requested in upstream pull request #8.

ShadowEO reviewed (commented):
I've made the changes in my fork as requested. Please review at your leisure :D

andersevenrud reviewed (approved):
LGTM. I still need to take it for a quick test drive before merging :)
